examples/seismic/source.py

```diff
@@ -34,15 +36,67 @@ def __new__(cls, name, ntime=None, npoint=None, ndim=None,
             ntime = ntime or data.shape[0]
 
         # Create the underlying PointData object
-        obj = PointData(name=name, dimensions=[time, p_dim],
-                        npoint=npoint, nt=ntime, ndim=ndim,
-                        coordinates=coordinates, **kwargs)
+        obj = PointData.__new__(cls, name=name, dimensions=[time, p_dim],
+                                npoint=npoint, nt=ntime, ndim=ndim,
+                                coordinates=coordinates, **kwargs)
 
         # If provided, copy initial data into the allocated buffer
         if data is not None:
             obj.data[:] = data
         return obj
 
+    def __init__(self, *args, **kwargs):
+        if not self._cached():
+            super(PointSource, self).__init__(*args, **kwargs)
+
 
 Receiver = PointSource
 Shot = PointSource
+
+
+class RickerSource(PointSource):
+    """
+    Symbolic object to encapsulate a set of sources with a
+    pre-defined Ricker wavelet.
+
+    :param name: Name for the reuslting symbol
+    :param f0: Peak frequency for Ricker wavelet
+    :param time: Discretized values of time
+    :param ndim: Number of spatial dimensions
+    """
+
+    def __new__(cls, *args, **kwargs):
+        time = kwargs.get('time')
+        f0 = kwargs.get('f0')
+        npoint = kwargs.get('npoint', 1)
+        kwargs['ntime'] = len(time)
+        kwargs['npoint'] = npoint
+        obj = PointSource.__new__(cls, *args, **kwargs)
+        for p in range(npoint):
+            obj.data[:, p] = obj.wavelet(f0, time)
+        return obj
+
+    def __init__(self, *args, **kwargs):
+        if not self._cached():
+            super(RickerSource, self).__init__(*args, **kwargs)
+
+    def wavelet(self, f0, t):
+        """
+        Create Ricker wavelet with a peak frequency f0 at time t.
+
+        :param f0: Peak frequency
+        :param t: Discretized values of time
+        """
```

Comment:
@mloubout Can you please provide more appropriate dostrings for these? Also, how does our implementation differ from more general ones? Should things like wavelength or amplitude be configurable?